internal/tiltfile/tiltextension/github_fetcher.go

```diff
@@ -4,63 +4,60 @@ import (
 	"context"
 	"fmt"
 	"io/ioutil"
-	"net/http"
+	"os"
+	"path"
 	"time"
 
-	"github.com/google/go-github/v29/github"
-
-	pkgtiltextension "github.com/tilt-dev/tilt/pkg/tiltextension"
+	"github.com/tilt-dev/go-get"
 )
 
-type GithubFetcher struct {
+type Downloader interface {
+	RootDir() string
+	Download(pkg string) (string, error)
 }
 
-// TODO(dmiller): DI github
-// TODO(dmiller): DI HTTP client
-func NewGithubFetcher() *GithubFetcher {
-	return &GithubFetcher{}
+type TempDirDownloader struct {
+	rootDir string
 }
 
-const githubTemplate = "https://raw.githubusercontent.com/tilt-dev/tilt-extensions/%s/%s/Tiltfile"
-
-func (f *GithubFetcher) Fetch(ctx context.Context, moduleName string) (ModuleContents, error) {
-	client := github.NewClient(nil)
-	masterBranch, _, err := client.Repositories.GetBranch(ctx, "tilt-dev", "tilt-extensions", "master")
+func NewTempDirDownloader() (*TempDirDownloader, error) {
+	dir, err := ioutil.TempDir("", "tilt-extensions")
 	if err != nil {
-		return ModuleContents{}, err
+		return nil, err
 	}
-	headOfMaster := masterBranch.GetCommit()
-	masterSHA := headOfMaster.GetSHA()
+	return &TempDirDownloader{rootDir: dir}, nil
+}
 
-	err = pkgtiltextension.ValidateName(moduleName)
-	if err != nil {
-		return ModuleContents{}, err
-	}
-	c := &http.Client{
-		Timeout: 20 * time.Second,
-	}
+func (d *TempDirDownloader) RootDir() string {
+	return d.rootDir
+}
 
-	urlText := fmt.Sprintf(githubTemplate, masterSHA, moduleName)
-	resp, err := c.Get(urlText)
-	if err != nil {
-		return ModuleContents{}, err
-	}
+func (d *TempDirDownloader) Download(pkg string) (string, error) {
+	dlr := get.NewDownloader(d.rootDir)
+	return dlr.Download(pkg)
+}
 
-	defer resp.Body.Close()
+type GithubFetcher struct {
+	dlr Downloader
+}
 
-	if resp.StatusCode != http.StatusOK {
-		return ModuleContents{}, fmt.Errorf("error fetching Tiltfile %q: %v", urlText, resp.Status)
-	}
+func NewGithubFetcher(dlr Downloader) *GithubFetcher {
+	return &GithubFetcher{dlr: dlr}
+}
+
+func (f *GithubFetcher) CleanUp() error {
+	return os.RemoveAll(f.dlr.RootDir())
+}
 
-	body, err := ioutil.ReadAll(resp.Body)
+func (f *GithubFetcher) Fetch(ctx context.Context, moduleName string) (ModuleContents, error) {
+	dir, err := f.dlr.Download(path.Join("github.com/tilt-dev/tilt-extensions", moduleName))
 	if err != nil {
-		return ModuleContents{}, err
+		return ModuleContents{}, fmt.Errorf("Fetching tilt-extensions: %v", err)
 	}
 
 	return ModuleContents{
 		Name:              moduleName,
-		TiltfileContents:  string(body),
-		GitCommitHash:     masterSHA,
```

Comment:
DM was persisting this to make it easier to deal with versioning in the future, though I'm now unable to recall or find a writeup of what his plan was for it. Do you have confidence it's unimportant, or is it worth thinking about a bit / re-adding?

Reply:
go-get has a notion of versioning.  I think it's based on reading all the repo tags, looking at the ones that look like semver strings, and trying to make some intelligent decisions about the latest upgrade-able one.

It doesn't even let you get commit hashes, because it needs to be able to download from systems that don't have commit hashes.

I have real complaints about the way go-get does versioning. But I'd rather inherit those problems than trying to implement our own versioning from scratch.

Will leave a comment on this in the code